frontend/src/pages/external/redirectComponents/PipelinesSdkRedirects.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { useParams, useLocation, useNavigate } from 'react-router-dom';
+import { Button } from '@patternfly/react-core';
+import { experimentRunsRoute, globalPipelineRunDetailsRoute } from '~/routes';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import { useRedirect } from '~/utilities/useRedirect';
+import RedirectErrorState from '~/pages/external/RedirectErrorState';
+
+/**
+ * Handles redirects from Pipeline SDK URLs to internal routes.
+ *
+ * Matches and redirects:
+ * - Experiment URL: /external/pipelinesSdk/{namespace}/#/experiments/details/{experimentId}
+ * - Run URL: /external/pipelinesSdk/{namespace}/#/runs/details/{runId}
+ */
+const PipelinesSdkRedirects: React.FC = () => {
+  const { namespace } = useParams<{ namespace: string }>();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const createRedirectPath = React.useCallback(() => {
+    if (!namespace) {
+      throw new Error('Missing namespace parameter');
+    }
+
+    // Extract experimentId from hash
+    const experimentMatch = location.hash.match(/\/experiments\/details\/([^/]+)$/);
+    if (experimentMatch) {
+      const experimentId = experimentMatch[1];
+      return experimentRunsRoute(namespace, experimentId);
+    }
+
+    // Extract runId from hash
+    const runMatch = location.hash.match(/\/runs\/details\/([^/]+)$/);
+    if (runMatch) {
+      const runId = runMatch[1];
+      return globalPipelineRunDetailsRoute(namespace, runId);
+    }
+
+    throw new Error('The URL format is invalid.');
+  }, [namespace, location.hash]);
+
+  const [redirect, { loaded, error }] = useRedirect(createRedirectPath);
+
+  React.useEffect(() => {
+    redirect();
+  }, [redirect]);
```

Comment:
This model is often very weird. I'm still investigating what's going on here -- but typically you don't useEffect around a response from another hook -- typically that means you just do it in that custom hook.

Reply:
Why do you have a loading state if you invoke redirect immediately irrespective of it?

Reply:
So you are saying I should just do the redirect in the hook itself. Thats fine i can make that change. And the loading state doesnt make sense here but in some cases it would. For this one i can get rid of it